phpBB/phpbb/db/driver/driver.php

```diff
@@ -793,6 +804,98 @@ function sql_build_query($query, $array)
 		return $sql;
 	}
 
+
+	protected function _process_boolean_tree_first($operations_ary)
+	{
+		if ($operations_ary[0] !== 'AND' &&
+			$operations_ary[0] !== 'OR')
+		{
+			$operations_ary = array('AND', $operations_ary);
```

Comment:
That's not what this is for. This is for a specific case where the user only specifies a single array with the single element for the WHERE clause. I didn't find a cleaner way of dealing with it, unfortunately...

I'll comment that.